### PR TITLE
Fix : Fix `SigRequirements` for NFT Transfer using alias

### LINKED ISSUE #4883: Fix SigRequirements for CryptoTransfer with alias
In the second scenario
- 0.0.A is sending 0.0.B an NFT of type 0.0.T with a royalty fee with fallback
- If 0.0.A receives fungible value in the transaction, the fallback will not be charged and 0.0.B does not necessarily need to sign
- Else 0.0.B will be charged a fallback, and must sign
But currently we require B to sign and we throw `INVALID_SIGNATURE` when `0.0.A` is used **as alias and account** in same transaction.

Reproduction steps:
- Ensure the `0.0.A` sending the NFT (using alias format) receives fungible value using its accountID format
- Ensure the`0.0.B` has no other debit

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #4883 
- Checks if the account mentioned in transfer list of NFT transfer are aliases and validates if their key should be added to signees list.
- Changed `SigRequirements` and `CryptoTransferHandler`

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/sigs/metadata/lookups/AccountSigMetaLookup.java [hunk 1]
```diff
@@ -27,4 +28,6 @@ public interface AccountSigMetaLookup {
     SafeLookupResult<AccountSigningMetadata> safeLookup(AccountID id);
 
     SafeLookupResult<AccountSigningMetadata> aliasableSafeLookup(AccountID idOrAlias);
+
+    EntityNum unaliasedAccount(AccountID idOrAlias);
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/sigs/metadata/lookups/AccountSigMetaLookup.java:32]: Javadoc useful here?

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java [hunk 2]
```diff
@@ -1371,13 +1371,19 @@ && isAlias(party)
...
                 return true;
             }
         }
         for (final var transfers : op.getTokenTransfersList()) {
             for (final var adjust : transfers.getTransfersList()) {
-                if (adjust.getAmount() > 0 && adjust.getAccountID().equals(target)) {
+                final var unaliasedAccountNum =
+                        sigMetaLookup.unaliasedAccount(adjust.getAccountID(), null);
+                final var unaliasedTargetNum = sigMetaLookup.unaliasedAccount(target, null);
+                if (adjust.getAmount() > 0 && unaliasedAccountNum.equals(unaliasedTargetNum)) {
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java:1386]: It'd be nice not to duplicate the logic here, but not a huge deal

File: test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java [hunk 3]
```diff
@@ -224,14 +225,94 @@ public List<HapiSpec> getSpecsInSuite() {
...
+                                                                                                .get(),
+                                                                                        +2))
+                                                                        .addAccountAmounts(
+                                                                                aaWith(
+                                                                                        otherAccountId
+                                                                                                .get(),
+                                                                                        -2))))
+                                .signedBy(DEFAULT_PAYER, PARTY, "otheraccount")
+                                .via(NFT_XFER))
+                .then(getTxnRecord(NFT_XFER).logged());
```
Comment (mhess-swl) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:313]: Are there any assertions here?

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/sigs/metadata/lookups/AccountSigMetaLookup.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/sigs/metadata/lookups/AccountSigMetaLookup.java:32]: Done

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java:1386]: Agreed. Modified that

File: test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java (hunk 3 above)
Comment (Neeharika-Sompalli) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoTransferSuite.java:313]: I think just validating its `SUCCESS`. If we don't add any `hasKnownStatus(..` it by default checks success

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/metadata/StateChildrenSigMetadataLookup.java [hunk 4]
```diff
@@ -160,6 +143,14 @@ public SafeLookupResult<AccountSigningMetadata> aliasableAccountSigningMetaFor(
         }
     }
 
+    @Override
+    public EntityNum unaliasedAccount(AccountID idOrAlias, final @Nullable LinkedRefs linkedRefs) {
+        if (isAlias(idOrAlias)) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/metadata/StateChildrenSigMetadataLookup.java:148]: ⚠️ We need to follow the pattern [here](https://github.com/hashgraph/hedera-services/blob/ed1019a61010924c1f89e7cd264ec499b8d11e4c/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/metadata/StateChildrenSigMetadataLookup.java#L182) to update `LinkedRefs` with the fact we used this alias.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/metadata/StateChildrenSigMetadataLookup.java:148]: Fixed . Thanks @tinker-michaelj !

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java [hunk 5]
```diff
@@ -1371,20 +1371,26 @@ && isAlias(party)
...
-                if (adjust.getAmount() > 0 && adjust.getAccountID().equals(target)) {
+                if (isReceivingFungibleValue(adjust, target)) {
                     return true;
                 }
             }
         }
         return false;
     }
 
+    private boolean isReceivingFungibleValue(final AccountAmount adjust, final AccountID target) {
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java]: Fixed . Thanks @tinker-michaelj !

**Review by mhess-swl — DISMISSED** [review]

**Review by kimbor — DISMISSED** [review]

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java [hunk 6]
```diff
@@ -210,15 +199,25 @@ private void handleHbarTransfers(
     }
 
     private boolean receivesFungibleValue(
-            final AccountID target, final CryptoTransferTransactionBody op) {
+            final AccountID target, final CryptoTransferTransactionBody op, final ReadableAccountStore accountStore) {
         for (var adjust : op.getTransfers().getAccountAmountsList()) {
-            if (adjust.getAmount() > 0 && adjust.getAccountID().equals(target)) {
+            final var unaliasedAccount = accountStore.getAccount(adjust.getAccountID());
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java:204]: This block L204-209 is duplicated with L215-220, would be nice to extract.

**Review by tinker-michaelj — APPROVED** [review]
One nit, otherwise LGTM! 🥂

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java (hunk 6 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java:204]: yes , will do in the next PR for modularization.

**Pull request merged**